src/lang/tscript-lib/core.ts

```diff
@@ -957,43 +957,60 @@ export const core = {
 		},
 		alert: function (text) {
 			let s = TScript.toString.call(this, text);
-			if (!this.service.documentation_mode && this.service.alert)
-				this.service.alert(s);
+			if (!this.service.documentation_mode && this.service.alert) {
+				this.status = "dialog";
+				this.dialogResult = null;
+				this.service.alert(s).then(() => {
+					this.dialogResult = {
+						type: this.program.types[Typeid.typeid_null],
+						value: { b: null },
+					};
+				});
+			}
 			return {
 				type: this.program.types[Typeid.typeid_null],
 				value: { b: null },
 			};
 		},
 		confirm: function (text) {
 			let s = TScript.toString.call(this, text);
-			let ret = false;
-			if (!this.service.documentation_mode && this.service.confirm)
-				ret = this.service.confirm(s);
+			if (!this.service.documentation_mode && this.service.confirm) {
+				this.status = "dialog";
+				this.dialogResult = null;
+				this.service.confirm(s).then((result) => {
+					this.dialogResult = {
+						type: this.program.types[Typeid.typeid_boolean],
+						value: { b: result },
+					};
+				});
+			}
 			return {
 				type: this.program.types[Typeid.typeid_boolean],
-				value: { b: ret },
+				value: { b: false },
 			};
 		},
 		prompt: function (text) {
 			let s = TScript.toString.call(this, text);
-			let ret = null;
 			if (!this.service.documentation_mode && this.service.prompt) {
-				ret = this.service.prompt(s);
-				if (ret === null)
-					return {
-						type: this.program.types[Typeid.typeid_null],
-						value: { b: null },
-					};
-				else
-					return {
-						type: this.program.types[Typeid.typeid_string],
-						value: { b: ret },
-					};
-			} else
-				return {
-					type: this.program.types[Typeid.typeid_string],
-					value: { b: "" },
-				};
+				this.status = "dialog";
```

Comment:
```suggestion
				this.status = "dialog";
				this.service.statechanged?.(false);
```
Same as in `alert` and `confirm`